components/director/internal/domain/formationassignment/status_service.go

```diff
@@ -64,6 +53,33 @@ func (fau *formationAssignmentStatusService) UpdateWithConstraints(ctx context.C
 		return errors.Wrapf(err, "while enforcing constraints for target operation %q and constraint type %q", model.NotificationStatusReturned, model.PreOperation)
 	}
 
+	stateFromReport := notificationStatusReport.State
+	fa.State = stateFromReport
+
+	if isErrorState(model.FormationAssignmentState(stateFromReport)) {
+		assignmentError := json.RawMessage{}
+		if notificationStatusReport.Error != "" {
+			assignmentErrorWrapper := AssignmentErrorWrapper{AssignmentError{
+				Message:   notificationStatusReport.Error,
+				ErrorCode: ClientError,
```

Comment:
The error in the notificationStatusReport is populated by the Client in case they had an error while performing some operation on their end. We are not editing the error field ourselves.